### PR TITLE
[BUGFIX] Do not reuse DataHandler

### PR DESCRIPTION
**Pull request by linawolf**
This is a follow up of https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/4260 following discussions on Slack: https://typo3.slack.com/archives/C028JEPJL/p1716473533796529

Releases: main, 12.4, 11.5

### CODE REVIEW

**Review by sbuerk — DISMISSED** [review]

File: Documentation/ApiOverview/DataHandler/UsingDataHandler/Index.rst [hunk 1]
```diff
@@ -62,8 +62,16 @@ the :php:`$data` and :php:`$cmd` arrays correctly prior to these chunks of code.
...
 :ref:`DataHandler basics <datahandler-basics>` chapter.
 
-The :php:`DataHandler` class can be injected into the constructor via
-:ref:`dependency injection <DependencyInjection>`.
+..  warning::
+    A new :php:`DataHandler` object **should** be created using
+    :php:`GeneralUtility::makeInstance(DataHandler::class)` before each use.
+    It is a stateful service and has to be considered polluted after use. Do not
+    call :php:`DataHandler::start()` or :php:`DataHandler::process_datamap()`
+    multiple time on the same instance.
```
Comment (garvinhicking) [inline Documentation/ApiOverview/DataHandler/UsingDataHandler/Index.rst:70]:
```suggestion
    multiple times on the same instance.
```

**Review by garvinhicking — CHANGES_REQUESTED** [review]
I found some more places (can't change ATM, can check tomorrow):

https://github.com/TYPO3-Documentation/raw/screenshots/Classes/Core/Functional/Framework/DataHandling/Scenario/DataHandlerWriter.php

https://github.com/TYPO3-Documentation/raw/TYPO3CMS-Reference-CoreApi/Documentation/ApiOverview/Fal/UsingFal/_ExamplesFileFolder/_CreatingFileReference.php

https://github.com/TYPO3-Documentation/raw/TYPO3CMS-Reference-CoreApi/Documentation/ApiOverview/DataHandler/Database/_BasicUsage.php

https://github.com/TYPO3-Documentation/raw/t3docs-examples/Classes/Controller/ModuleController.php

**Comment by linawolf** [comment]
I think we can also do separate PRs about the other locations, They were not introduced by https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/4260

**Comment by garvinhicking** [comment]
Only 2 of the links affect this repo, I think we should bundle this. Can do that in 2-3hrs.

**Comment by linawolf** [comment]
I added a PR for the examples extension and the two other cases here. I don't think we need to worry about the screenshots project. It was an internal tool and is not used anymore. We should archive it

**Comment by garvinhicking** [comment]
Ack, thanks for taking care!

Following lollis and stefans advice here for merging.

**Review by garvinhicking — APPROVED** [review]

**Pull request merged**